### PR TITLE
Autoretry CR-1199 CR-1266

### PR DESCRIPTION
**Pull request by lawschlosser**
- support for auto-retry for preempted/failed tasks
autoretry_client_02
autoretry_client_03

### CODE REVIEW

File: conductor/submitter.py [hunk 1]
```diff
@@ -238,8 +242,15 @@ def applyDefaultSettings(self):
         # Set the radio button on for the start/end frames by default
         self.on_ui_start_end_rdbtn_toggled(True)
 
-        # Set the default instance type
-        self.setInstanceType(self.default_instance_type)
+        # Set the default preemptible pref
+        default_preemptible = CONFIG.get('preemptible')
+        if default_preemptible != None:
```
Comment (flebel) [inline conductor/submitter.py]: PEP8: `!= None` should be `is not None`.

File: conductor/submitter.py [hunk 2]
```diff
@@ -238,8 +242,15 @@ def applyDefaultSettings(self):
...
+            self.setPreemptibleCheckbox(default_preemptible)
+
+        # Set the default preemptible pref
+        default_preempted_autoretry_count = CONFIG.get('autoretry_policy', {}).get("preempted", {}).get("max_retries")
+        if default_preempted_autoretry_count != None:
```
Comment (flebel) [inline conductor/submitter.py]: PEP8: `!= None` should be `is not None`.

File: conductor/resources/submitter.ui [hunk 3]
```diff
@@ -527,48 +562,78 @@
...
-             </property>
-            </widget>
-           </item>
-           <item>
-            <widget class="QWidget" name="ui_resource_wgt" native="true">
+            <widget class="QWidget" name="ui_preemptible_wgt" native="true">
              <property name="toolTip">
-              <string>[optional] Specify a resource string to associate the Conductor job with</string>
+              <string>A preemptible instance is up to 60% less expensive, but may be terminated before it completes a task.
+Best to use when not performing critical work.</string>
```
Comment (WMDConductor) [inline conductor/resources/submitter.ui]: Can we reword this to say:  "A preemptible instance can be significantly less expensive, but may be terminated before it completes a task."

File: conductor/submitter.py (hunk 1 above)
Comment (lawschlosser) [reply conductor/submitter.py]: ha, that's what I was going to do, but figured you would say that "we're not comparing ids", see our discussion: https://github.com/AtomicConductor/conductor_ee/pull/172#discussion_r148578437

File: conductor/resources/submitter.ui (hunk 3 above)
Comment (lawschlosser) [reply conductor/resources/submitter.ui]: ha, I was just asking mac about this. Sure I can do that.  Figured vaguely defined savings wouldn't be terribly inspiring (60% means something to people.  10% maybe not so much.  Yet both could be described as "immense savings" )

File: conductor/submitter.py (hunk 2 above)
Comment (lawschlosser) [reply conductor/submitter.py]: addressed

File: conductor/submitter.py (hunk 1 above)
Comment (lawschlosser) [reply conductor/submitter.py]: addressed

File: conductor/resources/submitter.ui (hunk 3 above)
Comment (lawschlosser) [reply conductor/resources/submitter.ui]: addressed

**Comment by sozawa** [comment]
@lawschlosser - What happened to the Project Selector?  Can't submit jobs without that.

File: conductor/resources/submitter.ui [hunk 4]
```diff
@@ -527,48 +562,78 @@
...
+                 <string/>
+                </property>
+                <property name="isFileUserPref" stdset="0">
+                 <bool>true</bool>
                 </property>
                </widget>
               </item>
               <item>
-               <widget class="QComboBox" name="ui_project_cmbx">
-                <property name="sizePolicy">
```
Comment (sozawa) [inline conductor/resources/submitter.ui]: This section needs to be added back in, it looks like

Comment (sozawa) [reply conductor/resources/submitter.ui]: Looks like it was moved to the top of the screen.  Looks like it's probably expected.

Comment (sozawa) [reply conductor/resources/submitter.ui]: It jumped up to line 191

Comment (lawschlosser) [reply conductor/resources/submitter.ui]: I moved it to the top of the UI

Comment (lawschlosser) [reply conductor/resources/submitter.ui]: @WMDConductor do you want to weigh in on the new placement of the Project box (and reorganization of the instance information)?

Orig vs new:
submiiter-new_v_orig

Comment (WMDConductor) [reply conductor/resources/submitter.ui]: This gets my thumbs up - picking your project first and then setting your parameters / instance type is a reasonable "top-down" flow.
